packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -8,39 +8,20 @@ import { useRef } from '@wordpress/element';
  */
 import useRefEffect from '../use-ref-effect';
 
-/* eslint-disable jsdoc/valid-types */
-/**
- * @template T
- * @param {T} value
- * @return {import('react').MutableRefObject<T>} A ref with the value.
- */
-function useFreshRef( value ) {
-	/* eslint-enable jsdoc/valid-types */
-	/* eslint-disable jsdoc/no-undefined-types */
-	/** @type {import('react').MutableRefObject<T>} */
-	/* eslint-enable jsdoc/no-undefined-types */
-	// Disable reason: We're doing something pretty JavaScript-y here where the
-	// ref will always have a current value that is not null or undefined but it
-	// needs to start as undefined. We don't want to change the return type so
-	// it's easier to just ts-ignore this specific line that's complaining about
-	// undefined not being part of T.
-	// @ts-ignore
-	const ref = useRef();
-	ref.current = value;
-	return ref;
-}
+/** @typedef {(e: DragEvent) => void} WPDragEventHandler */
+/** @typedef {(e: MouseEvent) => void} WPDragEndHandler */
 
 /**
  * A hook to facilitate drag and drop handling.
  *
- * @param {Object}                  props             Named parameters.
- * @param {boolean}                 props.isDisabled  Whether or not to disable the drop zone.
- * @param {(e: DragEvent) => void}  props.onDragStart Called when dragging has started.
- * @param {(e: DragEvent) => void}  props.onDragEnter Called when the zone is entered.
- * @param {(e: DragEvent) => void}  props.onDragOver  Called when the zone is moved within.
- * @param {(e: DragEvent) => void}  props.onDragLeave Called when the zone is left.
- * @param {(e: MouseEvent) => void} props.onDragEnd   Called when dragging has ended.
- * @param {(e: DragEvent) => void}  props.onDrop      Called when dropping in the zone.
+ * @param {Object}                         props             Named parameters.
+ * @param {boolean}                        props.isDisabled  Whether or not to disable the drop zone.
+ * @param {WPDragEventHandler | undefined} props.onDragStart Called when dragging has started.
+ * @param {WPDragEventHandler | undefined} props.onDragEnter Called when the zone is entered.
+ * @param {WPDragEventHandler | undefined} props.onDragOver  Called when the zone is moved within.
+ * @param {WPDragEventHandler | undefined} props.onDragLeave Called when the zone is left.
+ * @param {WPDragEndHandler | undefined}   props.onDragEnd   Called when dragging has ended.
+ * @param {WPDragEventHandler | undefined} props.onDrop      Called when dropping in the zone.
```

Comment:
@dmsnell or @sarayourfriend  you're welcome to push directly if you have better ideas to fix typing. The basic idea is that these handlers should be defined, but I also want to be able to set the ref.current back to `undefined` or `null` to avoid the React error

Reply:
proposed updates in [c8712b6](https://github.com/WordPress/gutenberg/pull/39038/commits/c8712b633c570b47f17a1742d76910be81d16f75) that don't require altering the function argument types and lets us go back to the `null` that I think you really preferred over `undefined`

Reply:
Excellent thanks, I missed that those parens for the inlines make a difference!